src/redux/actions/applicationActions.ts

```diff
@@ -50,6 +53,38 @@ export function saveAppSettings(appSettings: IAppSettings): (dispath: Dispatch)
     };
 }
 
+/**
+ * Ensures that a valid security token is associated with the project, otherwise creates one
+ * @param project The project to validate
+ */
+export function ensureSecurityToken(appSettings: IAppSettings, project: IProject):
```

Comment:
Mentioned above, but let's have this action lookup the appSettings so we don't need to pass it in here.